src/pybind/mgr/smb/handler.py

```diff
@@ -878,19 +892,28 @@ def _check_share(
                 smb_earmark,
             )
         else:
+            parsed_earmark = _parse_earmark(earmark)
+
+            # Check if the top-level scope is not SMB
             if not earmark_resolver.check_earmark(
                 earmark, EarmarkTopScope.SMB
             ):
                 raise ErrorResult(
                     share,
-                    msg=f"earmark has already been set by {earmark.split('.')[0]}",
+                    msg=f"earmark has already been set by {parsed_earmark['scope']}",
                 )
-            # Check if earmark is set by same cluster
-            if earmark.split('.')[2] != share.cluster_id:
+
+            # Check if the earmark is set by a different cluster
+            if (
+                parsed_earmark['cluster_id']
+                and parsed_earmark['cluster_id'] != share.cluster_id
+            ):
                 raise ErrorResult(
                     share,
-                    msg=f"earmark has already been set by smb cluster {earmark.split('.')[2]}",
+                    msg=f"earmark has already been set by smb cluster "
```

Comment:
I expect a linter to complain about this first f string as it has no brackets. You can drop the f prefix here.